### ISSUE TITLE
QPopupEdit @hide event not generated

### ISSUE THREAD

**Issue by carueda** [quasarframework/quasar#2861]
### Software version

Quasar: 0.17.18
OS: MacOs 10.13.1
Browsers:  Chrome, Firefox

### JsFiddle (for Quasar v0.15+ only)

https://jsfiddle.net/carueda/rep15w3x/1/

### What did you get as the error?

`@hide` event not consistently generated from QPopupEdit

### What were you expecting?

`@hide` event always generated regardless of the concrete action triggering the closing of the QPopupEdit, which would be completely consistent to what is documented:

> `@hide - Triggered after closing popup.`

### What steps did you take, to get the error?

Exercise the jsFiddle above and click the field to open the QPopupEdit.

To facilitate reproduction/inspection, I've added the following event handlers:

```js
        @show="$q.notify('show')"
        @hide="$q.notify('hide')"
        @cancel="$q.notify('cancel')"
```
This is what happens (in Quasar 0.17.18 at least):

| modifying field | action             | `@hide` generated | `@cancel` generated |
|---------------- |--------            | ------------------| --------------------|
| yes/no          |click outside popup | yes | no |
| yes/no          |Press Escape        | no  | yes |
| no              |Click CANCEL        | no  | no |
| yes             |Click CANCEL        | no  | yes |
| yes/no          |Click SET           | no  | no |

In short, the `@hide` event is *only* generated when clicking outside of the QPopupEdit.  For consistency and developer convenience in general,  I would suggest always generating it regardless of the concrete action triggering the closing of the QPopupEdit.  Thanks!

**Comment by Hamdan85** [comment]
same thing on q-modal

**Comment by rstoenescu** [comment]
This should be fixed in v1 beta.
